modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/Processing.scala

```diff
@@ -83,7 +96,8 @@ object Processing {
    */
   private case class BatchAfterTransform(
     toBeInserted: List[EventWithTransform],
-    entities: NonAtomicFields.Result,
+    v2Entities: Vector[TypedTabledEntity],
```

Comment:
I wonder if we need to keep here both: v2 and legacy entities in separate fields?

Eventually in `BatchAfterTransform` I think we only need field like `Vector[Field]` and both v2/legacy have that structure. So we could extract necessary details from v2/legacy and merge it when we create `BatchAfterTransform`.

Like:

```
  private case class BatchAfterTransform(
    toBeInserted: List[EventWithTransform],
    entities: Vector[Field],
    origBatchBytes: Long,
    origBatchCount: Long,
    badAccumulated: ListOfList[BadRow],
    tokens: Vector[Unique.Token]
  )
```

Unless we need to have here `Vector[TypedTabledEntity]` and I missed something!

Reply:
Well spotted!  It compiles and the tests still pass, so you must be right.